### PR TITLE
More unit tests, plus partial fix for #801

### PR DESCRIPTION
**Pull request by bkmartinjr**
## Issue and/or context:

We need to increase unit tests

## Changes:
* Improved unit tests for SparseNDArray
* New unit tests for factory.py 
* Fixed a couple of minor bugs related to error handling:
    * Partial fix for #801 - check encoding version in reification. We still need to check it for other code paths
    * Correctly handle the case of missing object metadata (use `meta.get(key, None)` rather than `meta[key]`)

## Notes for Reviewer:
None

### CODE REVIEW

File: apis/python/src/tiledbsoma/factory.py [hunk 1]
```diff
@@ -58,10 +63,12 @@ def _construct_member(
     try:
         if object_type == "array":
             with tiledb.open(member_uri, ctx=context.tiledb_ctx) as A:
-                spec_name = A.meta[SOMA_OBJECT_TYPE_METADATA_KEY]
+                spec_name = A.meta.get(SOMA_OBJECT_TYPE_METADATA_KEY, None)
```
Comment (thetorpedodog) [inline apis/python/src/tiledbsoma/factory.py:66]: nit: `, None)` not needed; it’s already the default

File: apis/python/src/tiledbsoma/factory.py [hunk 2]
```diff
@@ -70,6 +77,10 @@ def _construct_member(
 
     if spec_name is None:
         raise SOMAError("internal error: spec_name was not found")
+    if encoding_version is None:
+        raise SOMAError("internal error: encoding_version not found")
+    if encoding_version != SOMA_ENCODING_VERSION:
+        raise SOMAError("Unsupported SOMA object encoding version")
```
Comment (thetorpedodog) [inline apis/python/src/tiledbsoma/factory.py:83]: Should this be a ValueError?

File: apis/python/tests/test_sparse_nd_array.py [hunk 3]
```diff
@@ -754,3 +756,63 @@ def test_sparse_nd_array_table_slicing(tmp_path, io, write_format, read_format):
...
+    with pytest.raises(ValueError):
+        next(a.read((slice(10, -2),)).tables())
+    with pytest.raises(TypeError):
+        next(a.read("hi").tables())
+    with pytest.raises(ValueError):
+        next(a.read(coords=()).tables())
+    with pytest.raises(ValueError):
+        next(a.read(coords=[]).tables())
+    with pytest.raises(ValueError):
+        next(a.read(coords=((slice(None), slice(None)))).tables())
```
Comment (thetorpedodog) [inline apis/python/tests/test_sparse_nd_array.py:818]: https://youtube.com/clip/Ugkxi9EpXTo55k3mOGXWfXMEDJB0uYbOiBTO

**Review by thetorpedodog — APPROVED** [review]
one minor thing (ok if left as is), one question. looks good!

File: apis/python/src/tiledbsoma/factory.py (hunk 2 above)
Comment (bkmartinjr) [reply apis/python/src/tiledbsoma/factory.py:83]: Maybe...I'm a bit unclear on our exception UX approach with code like this.  I'll change for now, as we definitely have some work to do in this area.

File: apis/python/tests/test_sparse_nd_array.py (hunk 3 above)
Comment (bkmartinjr) [reply apis/python/tests/test_sparse_nd_array.py:818]: yeah, will parameterized - got a bit carried away there!

**Pull request merged**